From 3dc42aa9b1e2e4c124af4e252520801cbefd7a5a Mon Sep 17 00:00:00 2001
From: Monk Liu <Monk.Liu@amd.com>
Date: Wed, 1 Nov 2017 16:12:39 +0800
Subject: [PATCH 3769/5855] Revert "drm/amd/scheduler:fix duplicate operation
 in entity fini"

fix memory leak.
This reverts commit d6951b49faa8447a6a77cdb1ef3346b1a1786d31.
because when entity_fini is interrupted the jobs in queue still
not processed with job_begin, so the finish_cb is not hooked
on sched fence, we still need manually do cleanups.

Change-Id: I6e17bfeeac85062bc52f1d51b9697852b084845c
Signed-off-by: Monk Liu <Monk.Liu@amd.com>
Reviewed-by: Chunming Zhou <David1.Zhou@amd.com>
Signed-off-by: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
---
 drivers/gpu/drm/amd/scheduler/gpu_scheduler.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/amd/scheduler/gpu_scheduler.c b/drivers/gpu/drm/amd/scheduler/gpu_scheduler.c
index d118caf..f119ac1 100755
--- a/drivers/gpu/drm/amd/scheduler/gpu_scheduler.c
+++ b/drivers/gpu/drm/amd/scheduler/gpu_scheduler.c
@@ -234,6 +234,8 @@ void amd_sched_entity_fini(struct amd_gpu_scheduler *sched,
                         amd_sched_fence_scheduled(s_fence);
                         fence_set_error(&s_fence->finished, -ESRCH);
                         amd_sched_fence_finished(s_fence);
+			fence_put(&s_fence->finished);
+			sched->ops->free_job(job);
                 }
 
         }
-- 
2.7.4

